src/components/text-editor/prosemirror-adapter/menu/menu-commands.ts

```diff
@@ -42,7 +42,7 @@ const setActiveMethodForNode = (
         const node = $from.node($from.depth);
 
         if (node && node.type.name === nodeType.name) {
-            if (nodeType.name === 'heading') {
+            if (nodeType.name === LevelMapping.Heading && level) {
                 return node.attrs.level === level;
```

Comment:
Ok, so:

- if `node && node.type.name === nodeType.name`, but `nodeType.name` is not `'heading'`, return `true`
- if `node && node.type.name === nodeType.name`, and `nodeType.name` _is_ `'heading'`, but `level` is falsy, return `false`
- if `node && node.type.name === nodeType.name`, and `nodeType.name` _is_ `'heading'`, and `level` is truthy, return `true` if `node.attrs.level` and `level` are the same

Correct? 😅

Reply:
Yes. 😅 

This set of conditions should never happen as a `heading` must have a `level`:
if `node && node.type.name === nodeType.name`, and `nodeType.name` is `'heading'`, but `level` is falsy, return false

Reply:
We determine the `node.type` of the `current node` to show it is `active`. If the `node.type` is a `'heading'`, the `level` must be the same as the one on the menu item that was clicked so the relevant menu item shows as selected (active).

Reply:
Ah, yeah, "setActiveMethodForNode". Ok, that makes sense then 👍 🙂